### DISCUSSION TITLE
how can the datagrid columns be dynamically banding

### DISCUSSION THREAD

**Discussion by bouyeijiang** [AvaloniaUI/Avalonia#11058] [Q&A] [answered]
Alternatively,it can be bound system.data.datatble to datagrid

**Comment by maxkatz6** [comment]
> how can the datagrid columns be dynamically banding

You can modify DataGrid.Columns property dynamically from the C# code. I.e. removing or adding columns will update the UI automatically.
Alternatively, you can set Column.IsVisible property in the same way.

> ,it can be bound system.data.datatable to datagrid

It's unlikely to happen.

**Comment by bouyeijiang** [comment]
how can i do that?Because avalonia not like the WPF DataGrid,it was can be bounding datatable for example DataGrid.ItemSource=DataTable.I have looked at the examples on the official website and there are no more usage methods available

**Reply by timunie** [reply, marked as answer]
I think there is DataGridCollectionView or something similar available. DataTable support is not there iirc (but I'm not 100% sure). You may also want to check out [TreeDataGrid](https://docs.avaloniaui.net/docs/controls/treedatagrid)

**Reply by bouyeijiang** [reply]
thats good idea,its solved

**Reply by timunie** [reply]
marked as answer for others ^^

**Reply by JakkuSakura** [reply]
I tried DataGridCollectionView but it tries to read a class, not a dict
